File: 프로그래머스/lv2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> citations) {
    
    int answer = 0;
    
    std::sort(citations.begin(), citations.end(), std::greater<>());
    for (int i = 0; i < citations.size(); ++i) {
        if (citations[i] < i + 1) {
            return i;
        }
    }
    
    return citations.size();
    /*
    
    std::sort(citations.begin(), citations.end(), std::greater<>());
    
    for (int h = citations[0]; h >= 0; h--) {
        int count = 0;
        for (int i = 0; i < citations.size(); i++) {
            if (citations[i] >= h) {
                count++;
            }
            if (count >= h) {
                answer = h;
                return answer;
            }
        }
    }
    return answer;
    */
}

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        Arrays.sort(citations);

        // 배열 뒤집기
        for (int i = 0; i < citations.length / 2; i++) {
            int temp = citations[i];
            citations[i] = citations[citations.length - 1 - i];
            citations[citations.length - 1 - i] = temp;
        }

        for (int i = 0; i <= citations.length; i++) {
            if ((i == 0 || citations[i-1] >= i) && (i == citations.length || citations[i] <= i)) {
                answer = i;
            }
        }
        return answer;
    }
}